syntax = "proto3";

package robinhood;

option java_multiple_files = true;
option java_package = "com.aaron.grpc.robinhood";

service RobinhoodService {
  rpc login (LoginRequest) returns (LoginResponse) {}
  rpc logout(LogoutRequest) returns (LogoutResponse){}
  rpc quote (QuoteRequest) returns (QuoteResponse) {}
  rpc buy (BuyRequest) returns (BuyResponse) {}
  rpc sell (SellRequest) returns (SellResponse) {}
  rpc getBalance(BalanceRequest) returns (BalanceResponse){}
  rpc autoBuy(AutoBuyRequest) returns (AutoBuyResponse){}
  rpc autoSell(AutoSellRequest) returns (AutoSellResponse){}
  rpc getCompany(CompanyRequest) returns (CompanyResponse){}
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message LoginResponse {
  bool success = 1;
  string message = 2;
}

message LogoutRequest {}
message LogoutResponse {
  bool success = 1;
  string message = 2;
}

message QuoteRequest {
  string ticker = 1;
}

message QuoteResponse {
  double price = 1;
  string message = 2;
}

message BuyRequest {
  string ticker = 1;
  double amount = 2;
}

message BuyResponse {
  bool success = 1;
  string message = 2;
}

message SellRequest {
  string ticker = 1;
  double amount = 2;
}

message SellResponse {
  bool success = 1;
  string message = 2;
}
message BalanceRequest{

}
message BalanceResponse{
  double balance = 1;
  string message = 2;
}

message AutoBuyRequest {
  string ticker = 1;
  double target = 2;
  double amount = 3;
}

message AutoBuyResponse {
  bool success = 1;
  string message = 2;
}

message AutoSellRequest {
  string ticker = 1;
  double target = 2;
  double amount = 3;
}

message AutoSellResponse {
  bool success = 1;
  string message = 2;
}

message CompanyRequest{
  string ticker = 1;
}

message CompanyResponse{
  string company = 1;
  string message = 2;
}